app/main.py

```diff
@@ -1,2 +1,53 @@
+from typing import Hashable
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.threshold = int(self.capacity * 2 / 3)
+        self.items = [[] for _ in range(self.capacity)]
+
+    def __len__(self) -> int:
+        return self.length
+
+    def __setitem__(self, key: Hashable, value: any) -> None:
+        hash_ = hash(key)
+        index_ = hash_ % self.capacity
+
+        while self.items[index_]:
+            if self.items[index_][0] == key and self.items[index_][2] == hash_:
```

Comment:
There is no need to use 'and' part
When the first part is True, the second part always will be True either, because hash_ can't be different If keys are equal

Reply:
Oh, you're right! Thank you!